packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1768,6 +1768,116 @@ abstract class RenderBox extends RenderObject {
     return 0.0;
   }
 
+  Map<BoxConstraints, Size>? _cachedDryLayoutSizes;
+
+  /// Returns the [Size] that this [RenderBox] would like to be given the
+  /// provided [BoxConstraints].
+  ///
+  /// The size returned by this method is guaranteed to be the same size that
+  /// this [RenderBox] computes for itself during layout given the same
+  /// constraints.
+  ///
+  /// This function should only be called on one's children. Calling this
+  /// function couples the child with the parent so that when the child's layout
+  /// changes, the parent is notified (via [markNeedsLayout]).
+  ///
+  /// Calling this function is expensive as it can result in O(N^2) behavior.
+  ///
+  /// Do not override this method. Instead, implement [computeDryLayout].
+  @mustCallSuper
+  Size getDryLayout(BoxConstraints constraints) {
+    bool shouldCache = true;
+    assert(() {
+      // we don't want the checked-mode intrinsic tests to affect
+      // who gets marked dirty, etc.
+      if (RenderObject.debugCheckingIntrinsics)
+        shouldCache = false;
+      return true;
+    }());
+    if (shouldCache) {
+      _cachedDryLayoutSizes ??= <BoxConstraints, Size>{};
+      return _cachedDryLayoutSizes!.putIfAbsent(constraints, () => computeDryLayout(constraints));
+    }
+    return computeDryLayout(constraints);
+  }
+
+  /// Computes the value returned by [getDryLayout]. Do not call this
+  /// function directly, instead, call [getDryLayout].
+  ///
+  /// Override in subclasses that implement [performLayout] or [performResize].
+  /// This method should return the [Size] that this [RenderBox] would like to
+  /// be given the provided [BoxConstraints].
+  ///
+  /// The size returned by this method must match the [size] that the
+  /// [RenderBox] will compute for itself in [performLayout] (or
+  /// [performResize], if [sizedByParent] is true).
+  ///
+  /// If this algorithm depends on the size of a child, the size of that child
+  /// should be obtained using its [getDryLayout] method.
+  ///
+  /// ### When the size cannot be known
+  ///
+  /// There are cases where render objects do not have an efficient way to
+  /// compute their size without doing a full layout. For example, the size
+  /// may depend on the baseline of a child (which is not available without
+  /// doing a full layout), it may be computed by a callback about which the
+  /// render object cannot reason, or the layout is so complex that it
+  /// is simply impractical to calculate the size in an efficient way.
+  ///
+  /// In such cases, it may be impossible (or at least impractical) to actually
+  /// return a valid answer. In such cases, the function should call
+  /// [debugCannotComputeDryLayout] from within an assert and and return a dummy
+  /// value of `const Size(0, 0)`.
+  @protected
+  Size computeDryLayout(BoxConstraints constraints) {
```

Comment:
The name feels a little odd: If I was someone seeing it for the first time, I might wonder if there was a wet layout too, and what dry refers to.

Maybe `computeDryRunLayout`? `computeDesiredLayout`? `computeAPrioriLayout`?

Reply:
I am not really happy about any of these names... I added a paragraph to the docs explaining why this is dry and the other layout is wet.

Reply:
Instead of the paragraph, why not change the name to `computeDryRunLayout`?  Seems more self-documenting.